sli/tokenutils.cc

```diff
@@ -97,13 +106,20 @@ template <>
 float
 getValue< float >( const Token& t )
 {
-  DoubleDatum* id = dynamic_cast< DoubleDatum* >( t.datum() );
-  if ( id == NULL )
-  { // we have to create a Datum object to get the name...
-    DoubleDatum const d;
-    throw TypeMismatch( d.gettypename().toString(), t.datum()->gettypename().toString() );
+  DoubleDatum* dd = dynamic_cast< DoubleDatum* >( t.datum() );
+  if ( dd )
+  {
+    return ( float ) dd->get();
+  }
+  IntegerDatum* id = dynamic_cast< IntegerDatum* >( t.datum() );
+  if ( id )
+  {
+    return ( float ) id->get();
   }
-  return ( float ) id->get();
+
+  // we have to create a Datum object to get the name...
+  DoubleDatum const d;
+  throw TypeMismatch( d.gettypename().toString(), t.datum()->gettypename().toString() );
 }
 
```

Comment:
Se if `setValue< float >` below can be removed as well.

Reply:
Same as above.